src/main/java/liuni/services/TwitterService.java

```diff
@@ -65,22 +66,29 @@ public void setTwitterAccountConfig(TwitterAccountConfig config) {
     public TwitterTweetModel postStatus(String text) throws TwitterException {
         boolean isOkToPost = textErrorCheck(text);
         if (isOkToPost) {
-            Status status = twitter.updateStatus(text);
-            logger.info("Successfully updated status to [" + status.getText() + "].");
-            TwitterTweetModel tweet = getTweet(status);
+            TwitterTweetModel tweet = Stream.of(twitter.updateStatus(text))
+                                        .peek(status -> logger.info("Successfully updated status to [" + status.getText() + "]."))
+                                        .map(status -> getTweet(status))
+                                        .findFirst()
+                                        .get();
             return tweet;
         }
         return null;
     }
 
     public List<TwitterTweetModel> getTimeline() throws TwitterException {
         List<TwitterTweetModel> list = new ArrayList<TwitterTweetModel>();
```

Comment:
is list still needed?

Reply:
The way I understood it, yes because it is telling what list to put the collection into. But I will look into it more and verify that